src/LanguageServer.ts

```diff
@@ -543,6 +550,22 @@ export class LanguageServer {
         return item;
     }
 
+    private async onCodeAction(params: CodeActionParams) {
+        //ensure programs are initialized
+        await this.waitAllProgramFirstRuns();
```

Comment:
These awaits will potentially accumulate code action requests - should there instead be a mechanism to ignore codeactions when the file isn't ready? We may "miss" some actions but just moving the cursor around will produce then again.

Reply:
The language server protocol might already account for this and prevent additional requests if the first one hasn't arrived yet. So I'll test that first, and if we DO get multiple requests at the same time, then I'll add some code to ignore code actions that arrive before the programs are finished with their first runs.

Reply:
I just tested, and I can't get these to queue up. There's a `waitAllProgramFirstRuns()` in the onInitialized, so the vscode doesn't even send codeAction events until that's resolved, so this line is really a "just in case" measure.